source/Cosmos.System2/FileSystem/Disk.cs

```diff
@@ -221,7 +227,10 @@ public void DeletePartition(int index)
             {
                 mbr[i] = 0;
             }
-            Host.WriteBlock(0, 1, ref mbr);
+            Host.WriteBlock(0, 1, ref mbr); 
+            var part = parts[index];
+            Partition.Partitions.Remove(part.Host);
+            parts.RemoveAt(index);
```

Comment:
Is it possible to add tests for the new remove behavior to one of the existing fs related test kernels?

Reply:
I added a test for this behavor